src/detect-engine.c

```diff
@@ -4297,7 +4305,7 @@ static int DetectEngineTest02(void)
     if (de_ctx == NULL)
         goto end;
 
-    result = (de_ctx->inspection_recursion_limit == -1);
+    result = (de_ctx->inspection_recursion_limit == 3000);
```

Comment:
Why this value ?

Reply:
One of the failing tests was this. Now that we are checking the value of `inspection_resursion_limit` with `ByteExtract*`, whenever the retval is `< 0`, we reset the `inspection_recursion_limit` to its default compared to 0 earlier.

Reply:
So, if I understand correctly, should we not set `result` when `ByteExtract` fails and not use `de_ctx->inspection_recursion_limit` anymore ?

Otherwise, we should at least use `DETECT_ENGINE_DEFAULT_INSPECTION_RECURSION_LIMIT` instead of plain `3000`